src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessibility.java

```diff
@@ -565,8 +566,9 @@ public void run() {
                 if (pac == null) return;
                 AccessibleSelection as = pac.getAccessibleSelection();
                 if (as == null) return;
-                if (parent instanceof JList) {
-                    ((JList) parent).setSelectedIndex(i);
+                AccessibleTable pat = pac.getAccessibleTable();
+                if (pat instanceof AccessibleList) {
+                    ((AccessibleList) pat).setSelectionInterval(i, i);
                     return;
                 }
                 as.addAccessibleSelection(i);
```

Comment:
I would like to clarify one initial question. Why we cannot use `clearAccessibleSelection `to clean the current selection and then add a new one by the `addAccessibleSelection`?

Reply:
We tried to do this, it leads to an endless repetition of the selected line until it freezes.

Reply:
Probably because of the endless notifications loop? implementation of addSelectionInterval looks similar to setSelectionInterval. In some cases the addSelectionInterval just call setSelectionInterval.

Reply:
It looks like it is. Please formulate the question differently.

Reply:
It is unclear why the `setSelectionInterval `is actually any better than `addAccessibleSelection`. Both have a similar implementation,  in some cases, the addSelectionInterval just calls setSelectionInterval, and seems that posts similar notifications. So it is unclear why the setSelectionInterval works fine and addAccessibleSelection does not work.

Reply:
I tried calling clearSelection again instead of adding an interface:
```java
                if (as == null) return;
                if (pac.getAccessibleRole().equals(AccessibleRole.LIST)) {
                    as.clearAccessibleSelection();
                }
                as.addAccessibleSelection(i);
```
After that I changed the call to ```javax.swing.JList.AccessibleJList#addAccessibleSelection```
```java
JList.this.addSelectionInterval(i, i);
```
on the
```java
JList.this.getSelectionModel().setSelectionInterval(i, i);
```
Nothing changed. VoiceOVer still freezes.
The problem is not that setting the selection doesn't work, but that clearing the selection before setting causes the VO to fris.

Reply:
Some probably related points, the implementation of the addAccessibleSelection for the JComboBox. is the same:
```
            clearAccessibleSelection();
            JComboBox.this.setSelectedIndex(i);

```
It is possible that it also hangs?